dali/kernels/tensor_shape.h

```diff
@@ -515,6 +515,7 @@ struct TensorListShapeBase {
   void set_tensor_shape(int64_t sample, const SampleShape &sample_shape) {
     detail::check_compatible_ndim<sample_ndim, compile_time_size<SampleShape>::value>();
     assert(static_cast<int>(dali::size(sample_shape)) == static_cast<int>(sample_dim()));
+    assert(sample < nsamples && "Sample index happens to be greater than possible");
```

Comment:
I cannot deny the charm of this way of stating the error, but still consider "Sample index out of range." to be sufficient.
By the way - if you check it, you may also want to do this:
```suggestion
    assert(sample >= 0 && sample < nsamples && "Sample index out of range");
```

Reply:
:+1: